test/e2e/ingress.go

```diff
@@ -0,0 +1,311 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"sort"
+	"time"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// This test exercises the GCE L7 loadbalancer controller cluster-addon. It
+// will fail if the addon isn't running, or doesn't send traffic to the expected
+// backend. Common failure modes include:
+// * GCE L7 took too long to spin up
+// * GCE L7 took too long to health check a backend
+// * Repeated 404:
+//   - L7 is sending traffic to the default backend of the addon.
+//   - Backend is receiving /foo when it expects /bar.
+// * Repeated 5xx:
+//   - Out of quota (describe ing should show you if this is the case)
+//   - Mismatched service/container port, or endpoints are dead.
+
+var (
+	appPrefix         = "foo-app-"
+	pathPrefix        = "foo"
+	testImage         = "bprashanth/n-way-http:1.0"
+	httpContainerPort = 8080
+
+	// Average duration of loadbalancer startup.
+	avgLBLatency = 7 * time.Minute
+
+	// On average it takes ~6 minutes for a single backend to come online.
+	// We *don't* expect this poll to consistently take 15 minutes for every
+	// Ingress as GCE is creating/checking backends in parallel, but at the
+	// same time, we're not testing GCE startup latency. So give it enough
+	// time, and fail if the average is too high.
+	lbPollTimeout  = 15 * time.Minute
+	lbPollInterval = 30 * time.Second
+
+	// One can scale this test by tweaking numApps and numIng, the former will
+	// create more RCs/Services and add them to a single Ingress, while the latter
+	// will create smaller, more fragments Ingresses.
+	numApps = 6
```

Comment:
No, I was referring to numApps.  Why did you choose '6', as opposed to any other number? If someone came along and wanted to change the numbers, what would they need to know to do it intelligently?